src/linter.rs

```diff
@@ -23,9 +23,11 @@ use swc_ecmascript::parser::Syntax;
 
 pub use swc_common::SourceFile;
 
+#[derive(Default)]
```

Comment:
Why is `Default` needed?

Reply:
This `default` is for creating `Context` like this:

```rust
    let mut context = Context {
      file_name,
      source_map: self.ast_parser.source_map.clone(),
      leading_comments: leading,
      trailing_comments: trailing,
      ignore_directives: RefCell::new(ignore_directives),
      scope,
      control_flow,
      top_level_ctxt,
      ..Default::default() // <==============
    };
```

But it is not necessarily needed if `plugin_codes` field is not necessary in `Context`. So I'll deal with it after the other review comments are addressed.